src/bases.jl

```diff
@@ -444,10 +444,26 @@ function get_basis_diagonalizing end
 function _get_basis(M::AbstractManifold, p, B::DefaultOrthonormalBasis)
     return get_basis_orthonormal(M, p, number_system(B))
 end
+
```

Comment:
I think this should be the same as the `eltype` for tangent vectors? If we do have that, that is.
So if we don't I would maybe prefer to introduce that?

Reply:
How would you use `eltype` for tangent vectors? This is `eltype` for coordinates. In the current Quaternion PR I got quaternions as coordinates where real numbers were expected and this is supposed to fix the issue.

Reply:
Ah then I find the name misleading, since I thought it is the eltype used for the tangent vectors not the coordinates.

So if possible I would prefer these functions for points/vectors/coordinates? Then basis can use the ones it needs (for vectors/coordinates).

Reply:
BTW, the Quaternionic Unitary manifold is a counterexample, eltypes of vectors and coordinates are different (vectors are quaternions while coordinates are real numbers).

Reply:
that is why I propose to use three eltype functions :)

Reply:
It just needs the function for coordinates, other eltypes are determined automatically. We could maybe call it something like `coordinate_eltype`?

Reply:
We don't currently need functions that return point and tangent vector eltypes.

Reply:
Ok, then let's rename this (because from the name vector/coordinate is not clear). And when we need, we can add the vector eltype function somewhen later).